### PR TITLE
T40780 Split requirements from `requirements-dev.txt` file

### PR DESCRIPTION
**Pull request by JenySadadia**
Split `requirements-dev.txt` file into `requirements-tests.txt` and `requirements-linter.txt` files.
Add testing specific requirements to the `requirements-tests.txt` file.
Add linter specific requirements to `requirements-linter.txt`.

### CODE REVIEW

**Comment by mgalka** [comment]
Why this kind of split is needed?
In general the idea of having `requirements.txt` and `requirements-dev.txt` was to split the dependencies to those essential for running the application and those that need to be installed on dev machine to run the app, tests and do everything needed for the development. Is there a need to split it further to `-tests` and `-linter`?

**Comment by JenySadadia** [comment]
> Why this kind of split is needed? In general the idea of having `requirements.txt` and `requirements-dev.txt` was to split the dependencies to those essential for running the application and those that need to be installed on dev machine to run the app, tests and do everything needed for the development. Is there a need to split it further to `-tests` and `-linter`?

Please see this thread https://github.com/kernelci/kernelci-api/pull/216#pullrequestreview-1307250908

**Review by gctucker — CHANGES_REQUESTED** [review]
Please rework this to have:

* `requirements.txt` with the base requirements to just run the API
* `requirements-tests.txt` with added packages to run unit tests
* `requirements-dev.txt` with all the other packages (i.e. linters)

This way, `requirements.txt` and `requirements-dev.txt` can still be used as before and that's the standard way of doing things.  Then for the GitHub action to add linter annotations, we can use `requirements-tests.txt` (or any arbitrary name).

**Comment by gctucker** [comment]
Added `staging-skip` as this is breaking the Docker image build on staging.

**Comment by JenySadadia** [comment]
> Please rework this to have:
> 
> * `requirements.txt` with the base requirements to just run the API
> * `requirements-tests.txt` with added packages to run unit tests
> * `requirements-dev.txt` with all the other packages (i.e. linters)
> 
> This way, `requirements.txt` and `requirements-dev.txt` can still be used as before and that's the standard way of doing things. Then for the GitHub action to add linter annotations, we can use `requirements-tests.txt` (or any arbitrary name).

Fixed it.

File: README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Please follow the below instructions to test API endpoints.
 
 Install Python requirements with additional packages for testing:
 ```
-pip install -r docker/api/requirements-dev.txt
```
Comment (gctucker) [inline README.md:58]: I don't think this should change, the -dev file has all DW packages for development.

Comment (JenySadadia) [reply README.md:58]: Testing packages are in `requirements-tests.txt`.

Comment (gctucker) [reply README.md:58]: Yes but the standard way to install development packages is to install `requirements-dev.txt`.

Comment (JenySadadia) [reply README.md]: Yes, I dropped the commit.

**Comment by JenySadadia** [comment]
I'll add another commit for changing the requirements file for the `lint` job once we have https://github.com/kernelci/kernelci-api/pull/216 merged.

**Comment by gctucker** [comment]
@mgalka Do you have any other feedback on this?

**Review by gctucker — APPROVED** [review]

**Comment by mgalka** [comment]
> @mgalka Do you have any other feedback on this?

No, everything looks good

**Review by mgalka — APPROVED** [review]

**Pull request merged**